osquery/config/config.cpp

```diff
@@ -666,8 +666,6 @@ void Config::applyParsers(const std::string& source,
 
         auto doc = JSON::newFromValue(obj[key]);
         parser_config.emplace(std::make_pair(key, std::move(doc)));
-      } else {
-        parser_config.emplace(std::make_pair(key, JSON::newObject()));
```

Comment:
Why is this being removed? I'm not sure I follow.

Reply:
This is being removed because if you are using multiple config files, you get an empty object from all files that do not contain the a block for your config.

In ATC's case, this makes it impossible to tell if the ATC configuration was removed, or this is just another file that does not contain an ATC block.

The expected configuration for ATC should be an ATC block in one config file, so it knows what to remove when it changes. If an ATC block is removed entirely, it keeps tables, if the ATC block is emptied (left as an empty object) it removes all ATC tables.